### ISSUE TITLE
on shutdown the logger does not have enough time to increment the log counter

### ISSUE THREAD

**Issue by raymak** [raymak/contextualfeaturerecommender#250]
which leads to having startup and shutdown messages with the same number

**Comment by raymak** [comment]
have to check if unload is called for logger storage before logger.js

**Comment by gregglind** [comment]
isn't this what `unload.when` is for, which you already do?

**Comment by raymak** [comment]
yes, but on shutdown sometimes the promise won't get resolved. I think the solution would be to to use something like AsyncShutdown but it's not worth it for just a shutdown message.

**Comment by gregglind** [comment]
fwiw, the addon utils do it  and phone home with telemetry.

**Comment by raymak** [comment]
Not easy to reproduce. Happens only for some users and maybe once every 20 shutdowns.
